streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -338,6 +369,41 @@ abstract class DStream[T: ClassTag] (
   }
 
   /**
+   * Wrap a body of code such that the call site and operation scope
+   * information are passed to the RDDs created in this body properly.
+   */
+  protected def createRDDWith[U](time: Time)(body: => U): U = {
```

Comment:
This method is hard to name, but I don't think `withEnv` is a descriptive name at all. If you can come up with others, I am open to changing it.

I put it in its own method because we need to reuse it in `ForEachDStream`